CharlieBackend.Api.UnitTes/AccountServiceTests.cs

```diff
@@ -90,10 +92,155 @@ public async Task CreateAccountAsync()
+                IsActive = true,
+                Role = UserRole.Mentor
+            };
+
+            var notExistDto = new ChangeCurrentPasswordDto
+            {
+                Email = "notExist@exmaple.com"
+            };
+
+            var wrongPasswordDto = new ChangeCurrentPasswordDto
+            {
+                Email = "user@exmaple.com",
+                CurrentPassword = "wrongPassword",
+                NewPassword = newPassword,
+                ConfirmNewPassword = newPassword
+            };
+
+            var withoutSaltDto = new ChangeCurrentPasswordDto
+            {
+                Email = "withoutSalt@exmaple.com"
+            };
+
+            Account accountWithoutSalt = new Account
+            {
+                Id = 5,
+                IsActive = true,
+                Email = "withoutSalt@exmaple.com",
+                Password = HashPassword(oldPassword, salt),
+                Salt = null,
+                Role = UserRole.Mentor
+            };
+
+            _unitOfWorkMock.Setup(x => x.AccountRepository.GetAccountSaltByEmail(account.Email))
+                    .ReturnsAsync(salt);
+
+            _unitOfWorkMock.Setup(x => x.AccountRepository.GetAccountCredentialsByEmailAsync(changePass.Email))
+                    .ReturnsAsync(account);
+
+            _unitOfWorkMock.Setup(x => x.AccountRepository.GetAccountCredentialsByEmailAsync(accountWithoutSalt.Email))
+                    .ReturnsAsync(accountWithoutSalt);
+
+            var accountService = new AccountService(
+                _unitOfWorkMock.Object,
+                _mapper,
+                _notificationServiceMock.Object);
+
+            //Act
+            var notExistAccount = await accountService.ChangePasswordAsync(notExistDto);
+            var successResult = await accountService.ChangePasswordAsync(changePass);
+            var wrongPassword = await accountService.ChangePasswordAsync(wrongPasswordDto);
+            var accWithoutSalt = await accountService.ChangePasswordAsync(withoutSaltDto);
+
+            //Assert
+            Assert.NotNull(notExistDto);
+            Assert.NotNull(successResult);
+            Assert.NotNull(wrongPassword);
+            Assert.NotNull(accWithoutSalt);
+
+            Assert.Equal(ErrorCode.NotFound, notExistAccount.Error.Code);
+            Assert.Equal(ErrorCode.Conflict, wrongPassword.Error.Code);
+            Assert.Equal(ErrorCode.InternalServerError, accWithoutSalt.Error.Code);
+
+            Assert.Equal(updatedAccountDto.Id, successResult.Data.Id);
+            Assert.Equal(updatedAccountDto.Email, successResult.Data.Email);
+            Assert.Equal(updatedAccountDto.IsActive, successResult.Data.IsActive);
+            Assert.Equal(updatedAccountDto.Role, successResult.Data.Role);
+        }
+
         protected override Mock<IUnitOfWork> GetUnitOfWorkMock()
         {
             var mock = new Mock<IUnitOfWork>();
             return mock;
         }
+
+        #region Hash
```

Comment:
I didn't want this whole HashPassword logic here but without it it's impossible to make a successful test.
Without it, the condition "if (user.Password == checkPassword)" will never be true.